<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Tunneling</class>
 <widget class="QWidget" name="Tunneling">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>856</width>
    <height>385</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_4">
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_3">
     <item>
      <widget class="QLabel" name="label">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Maximum" vsizetype="Preferred">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="text">
        <string>Message code:</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QComboBox" name="mc">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer_6">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <widget class="QGroupBox" name="groupBox">
         <property name="title">
          <string>Control field</string>
         </property>
         <layout class="QGridLayout" name="gridLayout">
          <item row="0" column="0">
           <spacer name="horizontalSpacer">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeType">
             <enum>QSizePolicy::Fixed</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item row="0" column="1">
           <widget class="QLabel" name="label_7">
            <property name="text">
             <string>Frame type:</string>
            </property>
           </widget>
          </item>
          <item row="0" column="2">
           <widget class="QComboBox" name="frameType">
            <property name="currentIndex">
             <number>1</number>
            </property>
            <item>
             <property name="text">
              <string>Extended</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Standard</string>
             </property>
            </item>
           </widget>
          </item>
          <item row="3" column="1">
           <widget class="QLabel" name="label_11">
            <property name="text">
             <string>Priority:</string>
            </property>
           </widget>
          </item>
          <item row="3" column="2">
           <widget class="QComboBox" name="priority">
            <property name="currentIndex">
             <number>3</number>
            </property>
            <item>
             <property name="text">
              <string>System</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Normal</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Urgent</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Low</string>
             </property>
            </item>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QLabel" name="label_9">
            <property name="text">
             <string>Repeat:</string>
            </property>
           </widget>
          </item>
          <item row="1" column="2">
           <widget class="QComboBox" name="repeat">
            <property name="currentIndex">
             <number>1</number>
            </property>
            <item>
             <property name="text">
              <string>Yes</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>No</string>
             </property>
            </item>
           </widget>
          </item>
          <item row="4" column="1">
           <widget class="QLabel" name="label_10">
            <property name="text">
             <string>Acknowledge:</string>
            </property>
           </widget>
          </item>
          <item row="4" column="2">
           <widget class="QComboBox" name="acknowledge">
            <item>
             <property name="text">
              <string>Not requested</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Requested</string>
             </property>
            </item>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QLabel" name="label_8">
            <property name="text">
             <string>Broadcast:</string>
            </property>
           </widget>
          </item>
          <item row="2" column="2">
           <widget class="QComboBox" name="broadcast">
            <item>
             <property name="text">
              <string>System</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Domain</string>
             </property>
            </item>
           </widget>
          </item>
          <item row="5" column="1">
           <widget class="QLabel" name="label_12">
            <property name="text">
             <string>Confirm:</string>
            </property>
           </widget>
          </item>
          <item row="5" column="2">
           <widget class="QComboBox" name="confirm">
            <item>
             <property name="text">
              <string>No error</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Error</string>
             </property>
            </item>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer_2">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QVBoxLayout" name="verticalLayout_3">
       <item>
        <widget class="QGroupBox" name="groupBox_2">
         <property name="title">
          <string>Extended control field</string>
         </property>
         <layout class="QGridLayout" name="gridLayout_2">
          <item row="0" column="0">
           <spacer name="horizontalSpacer_4">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeType">
             <enum>QSizePolicy::Fixed</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item row="0" column="1">
           <widget class="QLabel" name="label_5">
            <property name="text">
             <string>Destination address type:</string>
            </property>
           </widget>
          </item>
          <item row="0" column="2">
           <widget class="QComboBox" name="destAddrType">
            <item>
             <property name="text">
              <string>Group</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Individual</string>
             </property>
            </item>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QLabel" name="label_4">
            <property name="text">
             <string>Hop count:</string>
            </property>
           </widget>
          </item>
          <item row="1" column="2">
           <widget class="QSpinBox" name="hopCount">
            <property name="maximum">
             <number>7</number>
            </property>
            <property name="value">
             <number>6</number>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QLabel" name="label_6">
            <property name="text">
             <string>Extended frame format:</string>
            </property>
           </widget>
          </item>
          <item row="2" column="2">
           <widget class="QComboBox" name="extFrameFormat">
            <item>
             <property name="text">
              <string>Standard</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>LTE</string>
             </property>
            </item>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </item>
     <item>
      <widget class="QGroupBox" name="additionalInfo">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="title">
        <string>Additional information</string>
       </property>
       <layout class="QGridLayout" name="gridLayout_3">
        <item row="0" column="2">
         <widget class="QComboBox" name="additionalInfoTypes">
          <property name="currentText">
           <string>PL Medium Information</string>
          </property>
          <property name="currentIndex">
           <number>1</number>
          </property>
          <item>
           <property name="text">
            <string>Reserved</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>PL Medium Information</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>RF Medium Information</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Busmonitor Status Info</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Timestamp Relative</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Time Delay Until Sending</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Extended Relative Timestamp</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>BiBat Information</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>RF Multi Information</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Preamble and Postamble</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>RF Fas tAcknowledgment</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Manufactor Specific Data</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Escape Code</string>
           </property>
          </item>
         </widget>
        </item>
        <item row="1" column="2">
         <widget class="QLineEdit" name="additionallnfoData">
          <property name="maxLength">
           <number>4</number>
          </property>
         </widget>
        </item>
        <item row="0" column="0">
         <spacer name="horizontalSpacer_5">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeType">
           <enum>QSizePolicy::Fixed</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item row="0" column="1">
         <widget class="QLabel" name="label_25">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Maximum" vsizetype="Preferred">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="text">
           <string>Type:</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QLabel" name="label_26">
          <property name="text">
           <string>Data (HEX):</string>
          </property>
         </widget>
        </item>
        <item row="2" column="1" rowspan="3" colspan="2">
         <widget class="QListWidget" name="additionalInfosList"/>
        </item>
        <item row="2" column="3">
         <widget class="QPushButton" name="addAdditionalInfo">
          <property name="text">
           <string>Add</string>
          </property>
         </widget>
        </item>
        <item row="3" column="3">
         <widget class="QPushButton" name="removeAdditionalInfo">
          <property name="text">
           <string>Remove</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_5">
     <item>
      <widget class="QLabel" name="label_14">
       <property name="text">
        <string>Source address:</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLineEdit" name="sourceAddress">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="text">
        <string>0.0.0</string>
       </property>
       <property name="maxLength">
        <number>9</number>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="label_15">
       <property name="text">
        <string>Destination address:</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLineEdit" name="destAddress">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="text">
        <string>0/0/0</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="label_16">
       <property name="text">
        <string>NPDU:</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLineEdit" name="npdu"/>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QTextEdit" name="textOuputTunneling">
     <property name="focusPolicy">
      <enum>Qt::NoFocus</enum>
     </property>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_4">
     <item>
      <widget class="QCheckBox" name="manualInput">
       <property name="text">
        <string>Manual input</string>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer_3">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeType">
        <enum>QSizePolicy::Fixed</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QLabel" name="label_17">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Maximum" vsizetype="Preferred">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="text">
        <string>cEmi frame (HEX):</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLineEdit" name="m_cemiFrame">
       <property name="enabled">
        <bool>true</bool>
       </property>
       <property name="text">
        <string/>
       </property>
       <property name="readOnly">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="tunnelingSendRequest">
       <property name="text">
        <string>Send Request</string>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer_2">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeType">
        <enum>QSizePolicy::Fixed</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QPushButton" name="connectTunneling">
       <property name="text">
        <string>Connect</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="disconnectTunneling">
       <property name="text">
        <string>Disconnect</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <tabstops>
  <tabstop>mc</tabstop>
  <tabstop>frameType</tabstop>
  <tabstop>priority</tabstop>
  <tabstop>repeat</tabstop>
  <tabstop>acknowledge</tabstop>
  <tabstop>broadcast</tabstop>
  <tabstop>confirm</tabstop>
  <tabstop>destAddrType</tabstop>
  <tabstop>hopCount</tabstop>
  <tabstop>extFrameFormat</tabstop>
  <tabstop>additionalInfoTypes</tabstop>
  <tabstop>additionallnfoData</tabstop>
  <tabstop>additionalInfosList</tabstop>
  <tabstop>addAdditionalInfo</tabstop>
  <tabstop>removeAdditionalInfo</tabstop>
  <tabstop>sourceAddress</tabstop>
  <tabstop>destAddress</tabstop>
  <tabstop>npdu</tabstop>
  <tabstop>textOuputTunneling</tabstop>
  <tabstop>manualInput</tabstop>
  <tabstop>m_cemiFrame</tabstop>
  <tabstop>tunnelingSendRequest</tabstop>
  <tabstop>connectTunneling</tabstop>
  <tabstop>disconnectTunneling</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>manualInput</sender>
   <signal>clicked(bool)</signal>
   <receiver>label</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>16</x>
     <y>364</y>
    </hint>
    <hint type="destinationlabel">
     <x>23</x>
     <y>23</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>manualInput</sender>
   <signal>clicked(bool)</signal>
   <receiver>mc</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>37</x>
     <y>363</y>
    </hint>
    <hint type="destinationlabel">
     <x>94</x>
     <y>21</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>manualInput</sender>
   <signal>clicked(bool)</signal>
   <receiver>groupBox</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>58</x>
     <y>361</y>
    </hint>
    <hint type="destinationlabel">
     <x>42</x>
     <y>121</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>manualInput</sender>
   <signal>clicked(bool)</signal>
   <receiver>groupBox_2</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>69</x>
     <y>362</y>
    </hint>
    <hint type="destinationlabel">
     <x>229</x>
     <y>116</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>manualInput</sender>
   <signal>clicked(bool)</signal>
   <receiver>sourceAddress</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>14</x>
     <y>359</y>
    </hint>
    <hint type="destinationlabel">
     <x>107</x>
     <y>247</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>manualInput</sender>
   <signal>clicked(bool)</signal>
   <receiver>label_14</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>56</x>
     <y>360</y>
    </hint>
    <hint type="destinationlabel">
     <x>17</x>
     <y>246</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>manualInput</sender>
   <signal>clicked(bool)</signal>
   <receiver>label_15</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>57</x>
     <y>364</y>
    </hint>
    <hint type="destinationlabel">
     <x>290</x>
     <y>246</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>manualInput</sender>
   <signal>clicked(bool)</signal>
   <receiver>destAddress</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>50</x>
     <y>357</y>
    </hint>
    <hint type="destinationlabel">
     <x>427</x>
     <y>245</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>manualInput</sender>
   <signal>clicked(bool)</signal>
   <receiver>label_16</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>29</x>
     <y>355</y>
    </hint>
    <hint type="destinationlabel">
     <x>577</x>
     <y>247</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>manualInput</sender>
   <signal>clicked(bool)</signal>
   <receiver>npdu</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>79</x>
     <y>355</y>
    </hint>
    <hint type="destinationlabel">
     <x>680</x>
     <y>250</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
